specs/interop/sequencer.md

```diff
@@ -30,11 +30,17 @@ without adding additional state-transition complexity or cross-chain synchronici
 ## Block Building
 
 The goal is to present information in a way where it is as efficient as possible for the block builder to only include
-executing messages that have a corresponding initiating message.
+executing messages that have a corresponding initiating message. It is not possible to enforce the ability to
+statically analyze a transaction, so execution MAY be required to determine the information required to include
+executing messages.
 
 ### Static analysis
 
-The block builder SHOULD use static analysis on executing messages to determine the dependency of the message.
+Note that static analysis is not always reliable, but it is far faster than having to perform
+execution to get the data required to validate an executing message.
+
+The block builder SHOULD use static analysis when possible on executing messages to determine
+the dependency of the message.
```

Comment:
When a transaction has a top level `to` address of the `CrossL2Inbox`, the call that it executes could result in reentering the `CrossL2Inbox` to relay a second executing message, and you cannot statically know this will occur. So you can use static analysis to identify invalid transactions and discard them to avoid executing those, but must execute them to confirm validity. Is that correct?

Reply:
That is actually a good point, this leads me to believe that static analysis can never work. Thank you for pointing this out